app/controllers/spree/paypal_controller_decorator.rb

```diff
@@ -0,0 +1,50 @@
+Spree::PaypalController.class_eval do
+  before_filter :enable_embedded_shopfront
+  before_filter :destroy_orphaned_paypal_payments, only: :confirm
+  after_filter :reset_order_when_complete, only: :confirm
+  before_filter :permit_parameters!
+
+  def cancel
+    flash[:notice] = Spree.t('flash.cancel', scope: 'paypal')
+    redirect_to main_app.checkout_path
+  end
+
+  # Clears the cached order. Required for #current_order to return a new order
+  # to serve as cart. See https://github.com/spree/spree/blob/1-3-stable/core/lib/spree/core/controller_helpers/order.rb#L14
+  # for details.
+  def expire_current_order
+    session[:order_id] = nil
+    @current_order = nil
+  end
+
+  private
+
+  def permit_parameters!
+    params.permit(:token, :payment_method_id, :PayerID)
```

Comment:
this is weird, do you need PayerId to be capitalized?

Reply:
https://github.com/openfoodfoundation/better_spree_paypal_express/blob/e28e4a8c5cedba504eea9cdad4be440d277d7e68/app/controllers/spree/paypal_controller.rb#L54

:man_shrugging:

Reply:
ok

That emoji is very useful man shrugging! :man_shrugging: